src/test/java/org/takes/misc/VerboseIteratorTest.java

```diff
@@ -0,0 +1,110 @@
+package org.takes.misc;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.Test;
+import org.hamcrest.MatcherAssert;
+
+import static org.hamcrest.Matchers.equalTo;
+
+import org.hamcrest.collection.IsIterableWithSize;
+
+/**
+ * Tests for {@link VerboseIterator}.
+ * @author marcus.sanchez (sanchez.marcus@gmail.com)
+ * @version $Id$
+ * @since 0.15.1
+ */
+public class VerboseIteratorTest {
+
+    /**
+     * VerboseIterator can return next value on a valid list.
+     */
+    @Test
+    public void returnsNextValue() {
+        MatcherAssert.assertThat(
+            new VerboseIterable<String>(
+                    Arrays.asList(
+                            "Accept: text/plain",
+                            "Accept-Charset: utf-8",
+                            "Authorization: Basic QWxhZGRpbjpvcGVuIHNlc2FtZQ==",
+                            "Cache-Control: no-cache",
+                            "From: user@example.com"
+                            ),
+                "Empty Error Message").iterator().next(),
+                equalTo("Accept: text/plain")
+            );
+    }
+
+    /**
+     * VerboseIterator can inform has a next value on a valid list.
+     */
+    @Test
+    public void informsHasNextValue() {
+        MatcherAssert.assertThat(
+                new VerboseIterable<String>(
+                        Arrays.asList(
+                                "Accept: text/plain",
+                                "Accept-Charset: utf-8",
+                                "Authorization: Basic QWxhZGRpbjpvcGVuIHNlc2FtZQ==",
+                                "From: user@example.com"
+                        ),
+                        "Empty Error Message"
+                    )
+                    .iterator().hasNext(), equalTo(true)
+       );
+    }
+
+    /**
+     * VerboseIterator next value throws exception on an empty list.
+     */
+    @Test(expected = RuntimeException.class)
+    public void nextValueThrowsExceptionOnEmptyList() {
+        new VerboseIterable<String>(
```

Comment:
@m4solutions yes, it is but when we call a mthor on the object that ends withy the bracket this is how it is done.